Sources/CvGame.cpp

```diff
@@ -3930,19 +3835,11 @@ bool CvGame::canTrainNukes() const
 		CvPlayer& kPlayer = GET_PLAYER((PlayerTypes)iI);
 		if (kPlayer.isAlive())
 		{
-			for (int iJ = 0; iJ < GC.getNumUnitClassInfos(); iJ++)
+			for (int iJ = 0; iJ < GC.getNumUnitInfos(); iJ++)
 			{
-				UnitTypes eUnit = (UnitTypes)GC.getCivilizationInfo(kPlayer.getCivilizationType()).getCivilizationUnits((UnitClassTypes)iJ);
```

Comment:
Is this check not necessary then? I have seen it in *many* places, but never looked into what it really means.

Reply:
C2C doesn't have civilization specific units, so regardless of what civ it is checking - a given unitclass type for - it will always return the same unit type for it.

Civ specific units depend upon the unitclass object; which we intend to remove eventually.